data/fin.lean

```diff
@@ -88,3 +91,23 @@ by cases i; refl
   {n} {C : fin (succ n) → Sort*} {H0 Hs} (i : fin n) :
   @fin.cases n C H0 Hs i.succ = Hs i :=
 by cases i; refl
+
+def nat_expand_type {n} (i : fin n) (k) : fin (n + k) :=
+⟨i.val,
+begin
+  have h : i.val < n, by apply i.is_lt,
+  apply lt_add_of_lt_of_nonneg,
```

Comment:
I'm sure you can reduce the size of this proof. For example, can you replace the `apply`s with direct application?